multisig/simulations/vip-002/vip-002-ethereum/simulations.ts

```diff
@@ -0,0 +1,569 @@
+  {
+    vTokens: ["vWBTC_Core"],
+    kink: "0.75",
+    base: "0",
+    multiplier: "0.09",
+    jump: "0.75",
+  },
+  {
+    vTokens: ["vWETH_Core"],
+    kink: "0.8",
+    base: "0",
+    multiplier: "0.09",
+    jump: "0.75",
+  },
+  {
+    vTokens: ["vUSDC_Core", "vUSDT_Core", "vcrvUSD_Core"],
+    kink: "0.8",
+    base: "0",
+    multiplier: "0.075",
+    jump: "0.8",
+  },
+  {
+    vTokens: ["vCRV_Core"],
+    kink: "0.45",
+    base: "0.02",
+    multiplier: "0.15",
+    jump: "3",
+  },
+  {
+    vTokens: ["vUSDC_Stablecoins", "vUSDT_Stablecoins", "vcrvUSD_Stablecoins", "vcrvUSD_Curve"],
+    kink: "0.8",
+    base: "0",
+    multiplier: "0.075",
+    jump: "0.8",
+  },
+  {
+    vTokens: ["vCRV_Curve"],
+    kink: "0.5",
+    base: "0.02",
+    multiplier: "0.2",
+    jump: "3",
+  },
+];
+
+const interestRateModelAddresses: { [key in VTokenSymbol]: string } = {};
+
+forking(19033343, () => {
+  let poolRegistry: Contract;
+
+  before(async () => {
+    poolRegistry = await ethers.getContractAt(POOL_REGISTRY_ABI, POOL_REGISTRY);
+  });
+
+  describe("Contracts setup", () => {
+    for (const [symbol, address] of Object.entries(vTokens) as [VTokenSymbol, string][]) {
+      checkVToken(address, vTokenState[symbol]);
+    }
+  });
+
+  describe("Post-Execution state", () => {
+    before(async () => {
+      await pretendExecutingVip(vip002());
+      for (const model of interestRateModels) {
+        for (const symbol of model.vTokens) {
+          const vToken = await ethers.getContractAt(VTOKEN_ABI, vTokens[symbol]);
+          interestRateModelAddresses[symbol] = await vToken.interestRateModel();
+        }
+      }
+    });
+    describe("PoolRegistry state", () => {
+      let registeredPools: { name: string; creator: string; comptroller: string }[];
+
+      before(async () => {
+        console.log(`PoolRegistry: ${poolRegistry.address}`);
+        registeredPools = await poolRegistry.getAllPools();
+      });
+
+      it("should have 3 pools", async () => {
+        expect(registeredPools).to.have.lengthOf(3);
+      });
+
+      it("should register Core pool in PoolRegistry", async () => {
+        const pool = registeredPools[0];
+        expect(pool.name).to.equal("Core");
+        expect(pool.creator).to.equal(ETHEREUM_MULTISIG);
+        expect(pool.comptroller).to.equal(COMPTROLLER_CORE);
+      });
+
+      it("should register Stablecoins pool in PoolRegistry", async () => {
+        const pool = registeredPools[1];
+        expect(pool.name).to.equal("Stablecoins");
+        expect(pool.creator).to.equal(ETHEREUM_MULTISIG);
+        expect(pool.comptroller).to.equal(COMPTROLLER_STABLECOINS);
+      });
+
+      it("should register Curve pool in PoolRegistry", async () => {
+        const pool = registeredPools[2];
+        expect(pool.name).to.equal("Curve");
+        expect(pool.creator).to.equal(ETHEREUM_MULTISIG);
+        expect(pool.comptroller).to.equal(COMPTROLLER_CURVE);
+      });
+
```

Comment:
nit: These could also be combined into one test without the `before` function. " Should register pools correctly".

I don't think it is necessary to have the variable registeredPools scoped outside of the tests and this would prevent it from being reached for when it doesn't need to be